commcare_connect/opportunity/helpers.py

```diff
@@ -0,0 +1,30 @@
+from django.db.models import BooleanField, Case, Count, DateTimeField, Max, Min, Q, When
+
+from commcare_connect.opportunity.models import Opportunity, OpportunityAccess
+
+
+def get_annotated_opportunity_access(opportunity: Opportunity):
+    access_objects = (
+        OpportunityAccess.objects.filter(opportunity=opportunity)
+        .select_related("user", "opportunityclaim")
+        .annotate(
+            last_visit_date_d=Max("user__uservisit__visit_date", filter=Q(user__uservisit__opportunity=opportunity)),
+            learn_progress_d=(Count("user__completed_modules") / Count("opportunity__learn_app__learn_modules")) * 100,
+            date_deliver_started=Min(
+                "user__uservisit__visit_date", filter=Q(user__uservisit__opportunity=opportunity)
+            ),
+            passed_assessment=Case(
```

Comment:
I think this filters for whether they have taken the assessment, not whether they have passed.

Reply:
Fixed. Added filter for passed assessments only. https://github.com/dimagi/commcare-connect/pull/213/commits/daff41020f739df242e773d68232b3958ff01042